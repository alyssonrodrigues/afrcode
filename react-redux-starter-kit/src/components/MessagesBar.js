import React from 'react'
import { withStyles } from 'material-ui/styles'
import Snackbar from 'material-ui/Snackbar'
import IconButton from 'material-ui/IconButton'
import CloseIcon from '@material-ui/icons/Close'
import { connect } from 'react-redux'

import styles from './MessagesBarStyles'
import { dismissMessage } from '../actions/messagesActions'

const MessagesBar = props => {
  const { classes } = props
  const { message } = props.messages
  const open = Boolean(message)
  const className = open ? classes[message.type] : null
  const msgText = open ? message.msgText : null
  return (
    <Snackbar
      className={className}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center'
      }}
      open={open}
      autoHideDuration={6000}
      onClose={props.dismissMessage}
      message={<span id='message-id'>{msgText}</span>}
      action={[
        <IconButton
          key='close'
          aria-label='Close'
          color='inherit'
          className={classes.close}
          onClick={props.dismissMessage}>
          <CloseIcon />
        </IconButton>
      ]}
    />
  )
}

const materialUIEnhance = withStyles(styles)(MessagesBar)

const mapStateToProps = ({ messages }) => ({ messages })

export default connect(mapStateToProps, { dismissMessage })(materialUIEnhance)
