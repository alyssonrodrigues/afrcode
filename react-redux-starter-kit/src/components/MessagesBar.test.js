import React from 'react'
import { shallow } from 'enzyme'

import { MessagesBar } from './MessagesBar'

describe('MessagesBar', () => {
  it('MessagesBar render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <MessagesBar classes={{}} messages={{message: 'OlÃ¡ mundo!'}} dismissMessage={jest.fn()} />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
