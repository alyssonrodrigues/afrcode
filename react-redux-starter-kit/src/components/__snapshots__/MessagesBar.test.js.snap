// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesBar MessagesBar render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={Object {}}
    dismissMessage={[Function]}
    messages={
        Object {
            "message": "OlÃ¡ mundo!",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ContentProps": Object {
        "elevation": 6,
        "style": Object {
          "backgroundColor": "#42a5f5",
        },
      },
      "action": Array [
        <WithStyles(IconButton)
          aria-label="Close"
          color="inherit"
          onClick={[Function]}
          style={
                    Object {
                              "padding": 8,
                            }
          }
>
          <pure(CloseIcon)
                    className={undefined}
          />
</WithStyles(IconButton)>,
      ],
      "anchorOrigin": Object {
        "horizontal": "center",
        "vertical": "bottom",
      },
      "autoHideDuration": 6000,
      "message": <span
        id="message-id"
/>,
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ContentProps": Object {
          "elevation": 6,
          "style": Object {
            "backgroundColor": "#42a5f5",
          },
        },
        "action": Array [
          <WithStyles(IconButton)
            aria-label="Close"
            color="inherit"
            onClick={[Function]}
            style={
                        Object {
                                    "padding": 8,
                                  }
            }
>
            <pure(CloseIcon)
                        className={undefined}
            />
</WithStyles(IconButton)>,
        ],
        "anchorOrigin": Object {
          "horizontal": "center",
          "vertical": "bottom",
        },
        "autoHideDuration": 6000,
        "message": <span
          id="message-id"
/>,
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
